Don't spawn a pen for uplink unless it's the selected choice

rdragan opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Traitors with no other pen won't have one spawned for them unless it's their uplink choice.
Fixes #48319 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Don't get meta'd for having a useless pen

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl: Thebleh
fix: Clown and Mime traitors don't get a pen unless it's their uplink choice
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

rdragan committed:
Don't spawn a pen for uplink unless it's needed

nemvar commented on code/datums/mind.dm:
```diff
@@ -317,7 +317,7 @@
 	if (!P) // If we couldn't find a pen in the PDA, or we didn't even have a PDA, do it the old way
 		P = locate() in all_contents
-		if(!P) // I do not have a pen.
+		if(!P && (traitor_mob.client.prefs.uplink_spawn_loc == UPLINK_PEN)) // I have no pen and I must write.
```
Move this whole section of code down by around 30 lines into the switch.
Specifically after line 345

rdragan replied:
Done. Thanks!

rdragan committed:
Spawn pen after checking uplink choice for pen

AnturK reviewed (changes requested):
Pen is a fallback in pda/radio options so you can't move it like this. Either do it early or change how the fallback location is chosen.

rdragan committed:
spawn pen uplink after exhausting other options

rdragan commented:
I've moved the spawning after everything so that it doesn't happen until we've exhausted all possible options. 

I've also removed the Pen option's fallback as it was previously unreachable anyway after we've gone through the trouble of spawning a pen. Traitors that don't have a pen but chose it as location will get one, along with traitors that have nothing. Leaving it in would mean Clowns/Mimes get their uplink in their PDA, effectively locking them out of the pen option. I think this is the best possible outcome.
